### PR TITLE
fix(ivy): remove query results from destroyed embedded views

### PR DESCRIPTION
**Pull request by pkozlowski-opensource**

### CODE REVIEW

File: packages/core/src/render3/util.ts [hunk 1]
```diff
@@ -127,7 +127,8 @@ export function isComponentDef<T>(def: DirectiveDef<T>): def is ComponentDef<T>
   return (def as ComponentDef<T>).template !== null;
 }
 
-export function isLContainer(value: RElement | RComment | LContainer | StylingContext): boolean {
+export function isLContainer(
+    value: RElement | RComment | LContainer | LView | StylingContext | null): boolean {
```
Comment (JoostK) [inline packages/core/src/render3/util.ts]: We can let TypeScript narrow the type of `value` after this call, would avoid some casts :)

```suggestion
    value: RElement | RComment | LContainer | LView | StylingContext | null): value is LContainer {
```

File: packages/core/src/render3/node_manipulation.ts [hunk 2]
```diff
@@ -242,14 +242,17 @@ export function destroyViewTree(rootView: LView): void {
   while (viewOrContainer) {
     let next: LView|LContainer|null = null;
 
-    if (viewOrContainer.length >= HEADER_OFFSET) {
+    if (isLContainer(viewOrContainer)) {
```
Comment (kara) [inline packages/core/src/render3/node_manipulation.ts]: Much clearer 👍

**Review by kara — APPROVED** [review]
LGTM

**Comment by kara** [comment]
[presubmit](http://test/OCL:231614131:BASE:231614168:1548871207408:60f774ca)

**Pull request closed without merging**